### PR TITLE
feat: extend API with datatype parameters

### PR DESCRIPTION
**Pull request by tpluscode**

### CODE REVIEW

**Comment by martinmaillard** [comment]
Nice! Will test.

Would it be possible to merge master into this branch to make it easier to test my own stuff on it? Thanks!

**Comment by tpluscode** [comment]
Sure, whenever you are ready. I fixed the tests so we should be good to go

**Comment by martinmaillard** [comment]
When I try to create an attribute, I get a validation (400) error with "Property template missing". Anything changed on that side? (I checked the payload and I *do* send propertyTemplate)

**Comment by tpluscode** [comment]
Nothing changed. And there are e2e tests for that. Please paste the payload

**Comment by martinmaillard** [comment]
Here's my payload:
```json
{
    "@type": "https://rdf-cube-curation.described.at/ValueAttribute",
    "https://rdf-cube-curation.described.at/propertyTemplate": "station-measuring-network",
    "https://rdf-cube-curation.described.at/column": "http://localhost:5678/project/b3b9270e-55db-4061-b14c-3d24b27bf756/source/udb0028-small-daten-de-csv/station-measuring-network"
}
```

And here's the response:
```json
{
    "type": "http://tempuri.org/DomainError",
    "title": "Cannot add attribute to table",
    "detail":"Property template missing",
    "status":400
}
```

**Comment by martinmaillard** [comment]
Seems to be working now

**Comment by martinmaillard** [comment]
I'm not sure I understand how `hydra:property datatype:parameters` and  `datatype:parameters rdfs:range dataCube:DatatypeParameters` works. Where is `dataCube:DatatypeParameters` defined?

What does the API expect? Is it a flat list of properties or can it be a nested object?

**Comment by martinmaillard** [comment]
Two things I noticed:
- The parameters don't seem to be provided by the API when retrieving the list of attributes
- After creating an attribute, "format" and "default" seem to have a string value "undefined" (literally). When I check the mapping:
    ```
    "csvw:default": "undefined",
    "csvw:format": "undefined",
    ```

File: api/lib/read-graphs/attribute.ts [hunk 1]
```diff
@@ -51,26 +41,22 @@ export async function getTableAttributes (tableId: string) {
...
+      ?attribute a dataCube:Attribute .
 
-      ?sourceColumn schema:name ?sourceName .
-      ?referencedColumn schema:name ?referencedName .
+      OPTIONAL {
+        ?attribute dataCube:columnMapping ?mapping .
+      }
+
+      OPTIONAL {
+        ?attribute <https://rdf-cube-curation.described.at/datatype/parameters> ?params .
```
Comment (martinmaillard) [inline api/lib/read-graphs/attribute.ts]: Could you also use the dataCube prefix here?

Comment (tpluscode) [reply api/lib/read-graphs/attribute.ts]: Yea, unfortunately not. `dataCube:datatype/parameters` is not a valid prefixed name.

Though I probably should add the `datatype` prefix which I used elsewhere

**Comment by martinmaillard** [comment]
Now I get the parameters, but the the datatype 😆

**Pull request merged**